### PR TITLE
fix(install): Use relative symlinks in deno install

### PR DESCRIPTION
**Pull request by nathanwhit**
Fixes https://github.com/denoland/deno/issues/25161

### CODE REVIEW

File: cli/npm/managed/resolvers/local.rs [hunk 1]
```diff
@@ -1048,10 +1048,13 @@ fn symlink_package_dir(
   // need to delete the previous symlink before creating a new one
   let _ignore = fs::remove_dir_all(new_path);
 
+  let old_path = crate::util::path::relative_path(new_parent, old_path)
+    .unwrap_or_else(|| old_path.to_path_buf());
+
   #[cfg(windows)]
   return junction_or_symlink_dir(old_path, new_path);
```
Comment (dsherret) [inline cli/npm/managed/resolvers/local.rs:1055]: Just remembered that junctions can't be relative. We might need to update this code to handle that.

Comment (dsherret) [reply cli/npm/managed/resolvers/local.rs:1055]: @piscisaureus recommends attempting to do a symlink first, then if an EPERM occurs to use junctions from then on.

Comment (nathanwhit) [reply cli/npm/managed/resolvers/local.rs:1055]: I guess they'll just have to be absolute on windows then, that's annoying

File: cli/npm/managed/resolvers/local.rs [hunk 2]
```diff
@@ -1048,41 +1048,48 @@ fn symlink_package_dir(
...
-          concat!(
-            "Failed creating junction and fallback symlink in node_modules folder.\n\n",
-            "{:#}\n\n{:#}",
-          ),
-          junction_err,
-          symlink_err,
-        ),
-      }
+    return junction::create(old_path, new_path)
+      .context("Failed creating junction in node_modules folder");
```
Comment (dsherret) [inline cli/npm/managed/resolvers/local.rs:1080]: Maybe add a comment about how junctions only support absolute paths?

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**